src/databricks/labs/ucx/assessment/aws.py

```diff
@@ -216,22 +230,33 @@ def _s3_actions(self, actions):
             s3_actions = [actions]
         return s3_actions
 
-    def _aws_role_trust_doc(self, external_id="0000"):
+    def _aws_role_trust_doc(self, self_assume_arn: str | None = None, external_id="0000"):
         return self._get_json_for_cli(
             {
                 "Version": "2012-10-17",
-                "Statement": [self._databricks_trust_statement(external_id)],
+                "Statement": [
+                    {
+                        "Effect": "Allow",
+                        "Principal": {
+                            "AWS": (
+                                [
+                                    "arn:aws:iam::414351767826:role/unity-catalog-prod-UCMasterRole-14S5ZJVKOTYTL",
+                                    self_assume_arn,
+                                ]
+                                if self_assume_arn
+                                else "arn:aws:iam::414351767826:role/unity-catalog-prod-UCMasterRole-14S5ZJVKOTYTL"
+                            )
+                        },
+                        "Action": "sts:AssumeRole",
+                        "Condition": {"StringEquals": {"sts:ExternalId": external_id}},
+                    }
+                ],
```

Comment:
this change will then just be 
```suggestion
"Statement": [self._databricks_trust_statement(external_id, self_assume_arn)],
```

Reply:
I removed _databricks_trust_statement it is unnecessary.